pkg/util/test/util.go

```diff
@@ -1958,78 +1928,93 @@ func validateCsiExtImages(cluster *corev1.StorageCluster, pxImageList map[string
+		if err := validateContainerImageInsidePods(cluster, image, "csi-resizer", &v1.PodList{Items: pods}); err != nil {
+			return err
+		}
 	} else {
 		return fmt.Errorf("failed to find image for csiResizer")
 	}
 
 	pxVer2_10, _ := version.NewVersion("2.10")
 	pxVersion, _ := version.NewVersion(getPxVersion(pxImageList, cluster))
 	if pxVersion.GreaterThanOrEqual(pxVer2_10) {
-		if value, ok := pxImageList["csiHealthMonitorController"]; ok {
-			csiHealthMonitorControllerImage = value
+		if image, ok := pxImageList["csiHealthMonitorController"]; ok {
+			if err := validateContainerImageInsidePods(cluster, image, "csi-external-health-monitor-controller", &v1.PodList{Items: pods}); err != nil {
+				return err
+			}
 		} else {
 			// CEE-452: csi-external-health-monitor-controller is removed from manifest, add back when resolved
 			logrus.Warnf("failed to find image for csiHealthMonitorController")
 		}
 	}
 
-	// Go through each pod and find all container and match images for each container
-	for _, pod := range pods {
-		for _, container := range pod.Spec.Containers {
-			if container.Name == "csi-external-provisioner" {
-				if container.Image != csiProvisionerImage {
-					return fmt.Errorf("found container %s, expected image: %s, actual image: %s", container.Name, csiProvisionerImage, container.Image)
-				}
-			} else if container.Name == "csi-snapshotter" {
-				if container.Image != csiSnapshotterImage {
-					return fmt.Errorf("found container %s, expected image: %s, actual image: %s", container.Name, csiSnapshotterImage, container.Image)
-				}
-			} else if container.Name == "csi-resizer" {
-				if container.Image != csiResizerImage {
-					return fmt.Errorf("found container %s, expected image: %s, actual image: %s", container.Name, csiResizerImage, container.Image)
-				}
-			} else if container.Name == "csi-health-monitor-controller" {
-				if container.Image != csiHealthMonitorControllerImage {
-					return fmt.Errorf("found container %s, expected image: %s, actual image: %s", container.Name, csiHealthMonitorControllerImage, container.Image)
-				}
-			}
-		}
-	}
 	return nil
 }
 
-func validateImageOnPods(image, namespace string, listOptions map[string]string) error {
-	pods, err := coreops.Instance().GetPods(namespace, listOptions)
-	if err != nil {
-		return err
+func validateContainerImageInsidePods(cluster *corev1.StorageCluster, image, containerName string, pods *v1.PodList) error {
+	logrus.Infof("Validating image for %s container inside pod(s)", containerName)
+
+	// Get PX Operator version
+	opVersion, _ := GetPxOperatorVersion()
+	if opVersion.GreaterThanOrEqual(opVer1_9_1) {
+		image = util.GetImageURN(cluster, image)
 	}
+
 	for _, pod := range pods.Items {
-		foundImage := false
-		for _, container := range pod.Spec.Containers {
-			if container.Image == image {
-				foundImage = true
-				break
+		foundContainer := false
+		foundImage := ""
+		if containerName == "init-cont" {
```

Comment:
let's not check this hardcoded name here, it's already checked inside of the loop in line 1986. instead you can add all containers and init containers in an array, check the array, it will also reduce duplicate code in if and else

Reply:
Done